src/ServicePulse.Host/app/js/views/monitored-endpoints/monitored-endpoints.controller.js

```diff
@@ -13,12 +13,9 @@
         });
 
         function updateUI() {
-            monitoringService.getRaw().then(function (data) {
-                $scope.endpoints = data["NServiceBus.Endpoints"];
-
-                timeoutId = $timeout(function () {
-                    updateUI();
-                }, 5000);
+            monitoringService.getEndpoints().subscribe(function (endpoint) {
+                $scope.endpoints = $scope.endpoints || {};
+                $scope.endpoints[endpoint.endpointName] = endpoint;
```

Comment:
That's exactly what this does.

Each request results in multiple endpoints, each endpoint is passed as a result to the subscribers. Each endpoint is rebound if it already exists, and is added to the object if it doesn't yet exist.

To be clear: and endpoint is a endpoint that is being monitored, not a monitoring service instance. So the graphs and label would be re-bound as it's updated.